models/src/main/kotlin/motif/models/Scope.kt

```diff
@@ -66,6 +67,18 @@ class ValidScope internal constructor(clazz: IrClass) : Scope(clazz) {
     override val factoryMethods: List<FactoryMethod> = objects?.factoryMethods ?: emptyList()
 
     override val dependencies: Dependencies? = Dependencies.fromScope(this)
+
+    companion object {
+
+        private fun detectScopeSuperinterface(scope: Scope, candidate: IrClass = scope.clazz) {
+            if (candidate != scope.clazz && candidate.hasAnnotation(motif.Scope::class)) {
```

Comment:
[nit]: I had a moment of confusion here due to my limited knowledge on Motif source code. I had thought that 
`if (candidate.hasAnnotation(motif.Scope::class))` would be good enough. Then I realized that  the `scope`  referred here is from motif source code, rather than the common Motif scopes defined by users. Hence it should be exempted.  If my speculation is correct,  we might consider define like this.
```if (candidate.hasAnnotation(motif.Scope::class) && !couldNotBeExempted(scope))```.

Reply:
Updated let me know if that makes more sense

Reply:
I reckon that this makes more sense~